tests/e2e_test.go

```diff
@@ -51,10 +59,12 @@ func (s *e2eSuite) SetupTest() {
 	})
 	s.writer = &MemWriter{}
 	s.app.Writer = s.writer
+
+	// noop exiter to prevent the app from exiting mid test
```

Comment:
urfave/cli that we use to run the app tends to exit on error, it should not happen according to [this](https://github.com/urfave/cli/pull/595), but since this behavior has been changing among different versions, I guess it's safe to assign a noop exiter, unless we want to test the app Exiting for sure. For some reason, without this the app either exits mid tests or it gets stuck forever 🤷‍♂️ I can dig into this further, but for now this is a work around that works.

Reply:
I think it's not a bug in urfave/cli but something we added in our wrapper. We exit the app unconditionally when any error occurs:

https://github.com/temporalio/cli/blob/a8bc242c033d2030026e7bd42bf34649347a9418/app/app.go#L68-L84

However, I'm actually not sure why we need to change this test at all for this PR?

Reply:
Nice find, yes that explains it! I think the suite is still a work in progress after my chat with @feedmeapples , but I agree that localizing the side-effects from app and server is a good idea. Let me fix that then.